File: src/main/java/org/funivan/lologa/algo/ga/genome/population/crossing/HalfCross.java
package org.funivan.lologa.algo.ga.genome.population.crossing;

import org.funivan.lologa.algo.ga.genome.GenomeInterface;

import java.util.HashMap;

@SuppressWarnings("unused")
public class HalfCross implements CrossingInterface {

    public HalfCross() {
    }

    @Override
    public HashMap<String, Double> cross(GenomeInterface father, GenomeInterface mother) {
        HashMap<String, Double> result = new HashMap<>();
        boolean takeMother = false;
        for (final String type : father.data().keySet()) {
            final GenomeInterface target = (takeMother ? mother : father);
            result.put(type, target.data().get(type));
            takeMother = !takeMother;
        }
        return result;
    }


}

File: src/main/java/org/funivan/lologa/algo/ga/genome/population/crossing/HalfCross.kt
package org.funivan.lologa.algo.ga.genome.population.crossing

import org.funivan.lologa.algo.ga.genome.GenomeInterface
import java.util.*

class HalfCross : CrossingInterface {

    override fun cross(father: GenomeInterface, mother: GenomeInterface): HashMap<String, Double> {
        val result = HashMap<String, Double>()
        var takeMother = false
        for (type in father.data().keys) {
            val target = if (takeMother) mother else father
            result.put(type, target.data()[type])
            takeMother = !takeMother
        }
        return result
    }


}
